libs/aws/langchain_aws/utils.py

```diff
@@ -108,6 +111,88 @@ def get_token_ids_anthropic(text: str) -> List[int]:
+        credentials_profile_name: str = None,
+        aws_access_key_id: SecretStr = None,
+        aws_secret_access_key: SecretStr = None,
+        aws_session_token: SecretStr = None,
+        endpoint_url: str = None,
+        config: Any = None,
+    ):
+    """Helper function to validate AWS credentials and create an AWS client."""
+
+    try:
+        import boto3
+
+        region_name = (
+            region_name
+            or os.getenv("AWS_REGION")
+            or os.getenv("AWS_DEFAULT_REGION")
+        )
+
+        client_params = {
+            "region_name": region_name,
+            "endpoint_url": endpoint_url,
+            "config": config,
+        }
+        client_params = {
+            k: v for k, v in client_params.items() if v
+        }
+
+        cred_params_provided = credentials_profile_name is not None or any([
+            aws_access_key_id, aws_secret_access_key, aws_session_token
+        ])
+
+        if cred_params_provided:
+            creds = {
+                "aws_access_key_id": aws_access_key_id,
+                "aws_secret_access_key": aws_secret_access_key,
+                "aws_session_token": aws_session_token,
+            }
+
+            if creds["aws_access_key_id"] and creds["aws_secret_access_key"]:
+                session_params = {
+                    k: v.get_secret_value() for k, v in creds.items() if v
+                }
+            elif any(creds.values()):
+                raise ValueError(
+                    f"If any of aws_access_key_id, aws_secret_access_key, or aws_session_token "
+                    f"are specified, then both aws_access_key_id and aws_secret_access_key "
+                    f"must be specified. Only received "
+                    f"{[(k, v) for k, v in creds.items() if v]}."
+                )
+            elif credentials_profile_name:
+                session_params = {"profile_name": credentials_profile_name}
+            else:
+                # Use default credentials
+                session_params = {}
+
+            session = boto3.Session(**session_params)
+
+            if not client_params.get("region_name") and session.region_name:
+                client_params["region_name"] = session.region_name
+
+            return session.client(service_name, **client_params)
+
+        return boto3.client(service_name, **client_params)
+
+    except UnknownServiceError as e:
+        raise ModuleNotFoundError(
+            f"Ensure that you have installed the latest boto3 package "
+            f"that contains the API for `{service_name}`."
+        ) from e
+    except BotoCoreError as e:
+        raise ValueError(
+            "Could not load credentials to authenticate with AWS client. "
+            "Please check that the specified profile name and/or its credentials are valid. "
+            f"Service error: {e}"
+        ) from e
+    except Exception as e:
+        raise ValueError(f"Error raised by service:\n\n{e}") from e
```

Comment:
Same as above, do we need to catch all exceptions, or just surface boto exceptions?

Reply:
This was also taken from the original classes' error handling, I don't have much context. But Pydantic [catches and wraps ValueError]((https://docs.pydantic.dev/latest/errors/validation_errors/#value_error)) as ValidationError, so I'm guessing this may have been done to make the base exceptions a little cleaner/more standardized for the user.